src/Docker.DotNet/Models/ServiceFilter.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+
+namespace Docker.DotNet.Models
+{
+    public class ServicesListParameters
+    {
+        public ServiceFilter Filters { get; set; }
+    }
+
+    public class ServiceFilter : Dictionary<string, string>
+    {
+        public string Id
+        {
+            get => this["id"];
+            set
+            {
+                if (ContainsKey("id"))
+                    this["id"] = value;
+                else
+                    Add("id", value);
+            }
+        }
+        public string Label
+        {
+            get => this["label"];
+            set
+            {
+                if (ContainsKey("label"))
+                    this["label"] = value;
+                else
+                    Add("label", value);
+            }
+        }
+        /// <summary> "replicated"|"global" </summary>
+        public string Mode

```

Comment:
Why not make this an enum with these values and do the parsing in the get/set but store in the dictionary as string so that we can have the type safety

Reply:
Change the order of your getter/setter and use an enum.
```C#
public ServiceCreationMode Mode
{
    get
    {
        ServiceCreationMode mode;
        Enum.TryParse(this["mode"], out mode);
        return mode;
    }
    set
    {
        this["mode"] = value.ToString()
    }
}
```